extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -284,20 +288,30 @@ export class ConsentUI {
     return placeholder;
   }
 
+  /**
+   * Get the client information that needs to be passed to cmp iframe
+   * @return {!Promise<JsonObject>}
+   */
+  getClientInfoPromise_() {
+    const consentStatePromise =
+        getServicePromiseForDoc(this.ampdoc_, CONSENT_STATE_MANAGER);
+    return consentStatePromise.then(consentStateManager => {
+      return consentStateManager.getConsentInstanceInfo().then(consentInfo => {
+        return dict({
+          'clientConfig': this.clientConfig_,
+          'consentState': consentInfo['consentState'],
```

Comment:
I decided to use 'consentState' and 'consentString', instead of a 'consentInfo' object because this leaves me more flexibility to change the internal `consentInfo` object format in the future. 

One thing I'd like to get your input is that: what do you think we should pass to the `consentState`. Right now, it's an enum value which essentially is a number. Do you think that we should keep a map and convert the enum value to a string?

Reply:
I very much like your idea of keeping this flexible, and passing parts of the consentInfo, as I am sure consent will evolve in the coming years, and require different information.

For the cosentState going from a number (0, 1, 2) to a string (accept, reject, dismiss), I definitely think we should use a string. It will make the code a lot more clear on their end, and isn't hard for us to do at all. Especially since they can simply build their own ENUMs and logic around a descriptive string, much easier than a number. I also feel like if we change/add new string values, it will probably be caught by an engineer, rather than someone scrolling through the documentation again 😄

Reply:
Sounds good. I added a helper method to convert the enum value to string (accepted/rejected/unknown)

This brings another question : )
The API we provide to CMP looks like 
`action: 'accept/reject/dismiss'`, but the consent state we pass back looks like `'accepted/rejected/unknown'`. Does this look confusing to any of you?

Reply:
Ummmm. To me it depends if unknown and dismiss mean the same thing?

If they don't I think that should be fine, because we are defining an action (present tense), and then we report back the state after the action has taken place (past tense).

If they do...Actually, I think distinguishing them still might be better. That way on the engineering side their is less confusion what was the action (request) vs. what is the state (response). Plus it would allow for more flexibility in the future, in case we ever want a state like "accepted, but now rejected" or something like that.

Reply:
Talked offline. We agreed it's somewhat confusing in both ways. But `dismiss` doesn't necessary equal to `unknown`, because there could be stored value. Because of this, we decided to go with `accepted/rejected/unknown` 

PTAL